pkg/sdk/extract.go

```diff
@@ -106,9 +106,26 @@ type ExtractRequest interface {
 	//  - sdk.FieldTypeIPNet: net.IPNet, *net.IPNet
 	SetValue(v interface{})
 	//
+	// SetValueOffsets sets the start offset and length of the field.
+	// The start offset must be from the beginning of the event to the
+	// start of the field data. sdk.PluginEventHeaderSize should be used
+	// to get the event header size. {0,0} can be used to indicate that
+	// the field doesn't correspond to any bytes in the event or log
+	// data. Offsets must be a slice of uint32s if IsList() is true, or
+	// single uint32s otherwise.
+	SetValueOffsets(start interface{}, length interface{})
```

Comment:
Wouldn't it be better to use a variadic function and pass a struct as an argument?

For example:
```go

type ValueOffset struct {
   Start uint32
   Len uint32
}

// ...

func SetValueOffsets(offsets ...ValueOffset)
```

@jasondellaluce wdyt?

Reply:
The variadic version of SetValueOffsets certainly looks cleaner. I switched it over and added some tests. I also updated github.com/falcosecurity/plugins/pull/677 accordingly.